tests/unit/test_placeholders_with_steps.py

```diff
@@ -387,3 +409,72 @@ def test_schema_validation_for_step_input():
                 "ParamB": "SampleValueB"
             }
         )
+
+def test_step_result_placeholder():
+    step_result = StepResult()
+
+    test_step_01 = Task(
+        state_id='StateOne',
+        result_selector={
+            'ParamA': step_result["foo"],
+            "ParamC": "SampleValueC"
+        }
+    )
+
+    expected_repr = {
+        "Type": "Task",
+        "ResultSelector": {
+            "ParamA.$": "$['foo']",
+            "ParamC": "SampleValueC"
+        },
+        "End": True
+    }
+
+    assert test_step_01.to_dict() == expected_repr
+
+def test_schema_validation_for_step_result():
+
+    step_result = StepResult(
+        schema={
+            "Payload": {
+                "Key01": str
+            }
+        }
+    )
+
+    with pytest.raises(ValueError):
+        test_step_01 = Task(
+            state_id='StateOne',
+            result_selector={
+                'ParamA': step_result["Payload"]["Key02"],
+                "ParamB": "SampleValueB"
+            }
+        )
+
+    with pytest.raises(ValueError):
+        test_step_02 = Task(
+            state_id='StateOne',
+            parameters={
+                'ParamA': step_result["Payload"].get("Key01", float),
```

Comment:
(same comment) nit: Let's use same notation for all keys for consistency - either `"` or `'`